File: Order-Management_System/src/main/java/com/example/demo/entity/Order.java
package com.example.demo.entity;


import java.sql.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.Builder;
//importing all the annotation from lombok for getter,setter and constructor
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
// we use a builder to create instances of our class.
@Builder
@Entity
// creates table in database with name ORDER_DETAILS
@Table(name = "ORDER_DETAILS")
public class Order {
// instance variable for customer entity
	
// order_id is auto generated
	@Id
	@GeneratedValue
	private Integer order_id;
	
	private String order_name;
	
	private int shipping_charge;
	
	private String status;
	
	private Date order_date;
	
	private int total_price;
	
// this connect with Customer entity using one to one mapping
// relationship
	
	@OneToOne(cascade = CascadeType.ALL)
	private Customer customer;

// this connect with Product entity using one to many mapping
// relationship	
	@OneToMany(cascade = CascadeType.ALL)
	private List<Product> product;
	
}

File: src/app/order.ts
import { Customer } from "./customer";
import { Product } from "./product";

export class Order {
    order_name:string='';
    shipping_charge:any;
    status:string='';
    total_price:any;
    order_date:any;
    customer:Customer={
        customer_name:'',
        customer_contact:'',
        customer_email:'',
        customer_address:'',
        pin:0
    };
    product:Product={
        product_name:'',
        product_quantity:0,
        product_price:0
    };
    
    // customer_contact:string='';
    // customer_email:string='';
    // customer_address:string='';
    // pin:any;
    
    // product_name:string='';
    // product_quantity:any;
    // product_price:any;
    // customer_name:string='';
    // customer_contact:string='';
    // customer_email:string='';
    // customer_address:string='';
    // pin:any;
    
    // product_name:string='';
    // product_quantity:any;
    // product_price:any;
    // customer_name:string='';
    // customer_contact:string='';
    // customer_email:string='';
    // customer_address:string='';
    // pin:any;
    
}
